specs/consensus.md

```diff
@@ -71,16 +85,149 @@ Consensus Rules
 
 ## Leader Selection
 
+TODO
+
 ## Fork Choice
 
+TODO
+
 ## Block Validity
 
-### State Transitions
+The validity of a newly-seen block, `block`, is determined by two components, detailed in subsequent sections:
+1. [Block structure](#block-structure): whether the block header is valid, and data in a block is arranged into a valid and matching data root (i.e. syntax).
+1. [State transition](#state-transitions): whether the application of transactions in the block produces a matching and valid state root (i.e. semantics).
+
+## Block Structure
+
+Before executing [state transitions](#state-transitions), the structure of the [block](./data_structures.md#block) must be verified.
+
+The following block fields are acquired from the network and parsed (i.e. [deserialized](./data_structures.md#serialization)). If they cannot be parsed, the block is ignored but is not explicitly considered invalid by consensus rules. Further implications of ignoring a block are found in the [networking spec](./networking.md).
+1. [block.header](./data_structures.md#header)
+1. [block.availableDataHeader](./data_structures.md#availabledataheader)
+1. [block.lastCommit](./data_structures.md#commit)
+
+If the above fields are parsed successfully, the available data `block.availableData` is acquired in erasure-coded form as [a list of share rows](./networking.md#availabledata), then parsed. If it cannot be parsed, the block is ignored but not explicitly invalid, as above.
```

Comment:
The whole data shouldn't need to be acquired due to DA proofs etc.

Reply:
For now the assumption is a full node fully downloads the entire block, so as to check everything (including that the erasure coding was done correctly). Later on we can specifically notate which functionality is required for each node type, as we define node types in #22.